vault/src/process.rs

```diff
@@ -0,0 +1,62 @@
+use std::{
+    fs::{self, File, OpenOptions},
+    io::Write,
+    path::PathBuf,
+    process,
+    str::FromStr,
+};
+
+use service::Error;
+use sysinfo::{Pid, System, SystemExt};
+
+pub fn pid_file_path(spec_name: String, account_id: String) -> PathBuf {
+    let file_name = format!("{}_{}.pid", spec_name, account_id);
+    // Use the default temporary directory of the OS
+    let mut path = std::env::temp_dir();
+    path.push(file_name);
+    path
+}
+
+pub fn try_create_pid_file(spec_name: String, account_id: String, sys: &mut System) -> Result<File, Error> {
+    let path = pid_file_path(spec_name, account_id);
+    if path.exists() {
+        let mut pid = fs::read_to_string(path.clone())?;
+        pid = pid.strip_suffix("\n").unwrap_or(&pid).to_string();
+
+        print!("{}", pid);
+        if sys.refresh_process(Pid::from_str(&pid)?) {
```

Comment:
PIDs can be reused by other processes, so we might want to improve robustness by checking the details of the process with that PID. At least the name, possibly also its arguments